model/repository/MonitoringRepository.php

```diff
@@ -460,23 +460,52 @@ protected function prepareOrderStmt($order): string
     {
         $order = explode(',', $order);
         $result = [];
-        $primaryTableColumns = $this->getPrimaryColumns();
-        foreach ($order as $ruleNum => $orderRule) {
-            preg_match('/([a-zA-Z_][a-zA-Z0-9_]*)\s?(asc|desc)?\s?(string|numeric)?/i', $orderRule, $ruleParts);
+        foreach ($order as $orderRule) {
+            $result[] = $this->buildSingleOrderRule($orderRule);
```

Comment:
`buildSingleOrderRule()` potentially returns an empty string now. Although, I understand it won't happen if the `$orderRule` is defined properly, I still suggest filtering the empty values here.
```suggestion
            $orderRule = $this->buildSingleOrderRule($orderRule);

            if ($rule) {
                $result[] = $orderRule;
            }
```